docs/user-guides/acme-dns-validation.md

```diff
@@ -0,0 +1,94 @@
+# Issuing an ACME certificate using DNS validation
+
+`cert-manager` can be used to obtain certificates from a CA using the [ACME][1] protocol. The ACME protocol supports various challenges which are used to prove ownership of a domain so that a valid certificate can be issued for that domain. One such challenge is `dns-01`. With the DNS challenge, you prove you own a domain by proving you control domain's DNS records. You do this by creating a TXT record with specific content.
+
+The following `Issuer` defines the necessary information to enable DNS validation. You can read more about the `Issuer` resource [here][2].
+
+```yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging
+  namespace: default
+spec:
+  acme:
+    # The ACME server URL
+    server: https://acme-staging.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: user@example.com
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    # ACME dns-01 provider configurations
+    dns01:
+      # Here we define a list of DNS-01 providers that can solve DNS challenges
+      providers:
+      - name: prod-dns
+        clouddns:
+          # A secretKeyRef to a google cloud json service account
+          serviceAccountSecretRef:
+            name: clouddns-service-account
+            key: service-account.json
+          # The project in which to update the DNS zone
+          project: gcloud-prod-project
+      - name: cf-dns
+        cloudflare:
+          email: user@example.com
+          # A secretKeyRef to a cloudflare api key
+          apiKeySecretRef:
+            name: cloudflare-api-key
+            key: api-key.txt
+```
+
+We have specified the ACME server URL for Let's Encrypt's [staging environment][3]. The staging environment will not issue trusted certificates but is used to ensure that the verification process is working properly before moving to production. Let's Encrypt's production environment imposes much stricter [rate limits][4], so to reduce the chance of you hitting those limits it is highly recommended to start by using the staging environment. To move to production, simply change the URL to `https://acme-v01.api.letsencrypt.org/directory`.
```

Comment:
```
To move to production, simply change the URL to `https://acme-v01.api.letsencrypt.org/directory`.
```

For now, changing the acme URI in an issuer won't *quite* magically switch all their certs, and is not something we test works in our automated tests. It may work at some point in the future, but for now the recommendation should be to create a second issuer `letsencrypt-prod` or the like.